### PR TITLE
#143412443 Set up front end unit testing

### PR DESCRIPTION
**Pull request by muchai-mercy**
#### What does this PR do?
Uses Jasmine and Karma for front end unit testing

 #### Description of Task to be completed?
npm install karma --save-dev
npm install jasmine --save-dev
npm install karma-jasmine --save-dev
npm install karma-chrome-launcher --save-dev
karma init
karma start karma.conf.js
Add configuration information to karma.conf.js file. i.e plugins, browser, frameworks, files, preprocessors, reporters

 #### How should this be manually tested?
karma start

 #### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2003755/stories/143412443

 #### Screenshots
screen shot 2017-04-25 at 3 27 39 pm

### CODE REVIEW

**Comment by gngchrs** [comment]
@Mercy-Muchai @Ewanjiru a few points
* See `https://github.com/tlvince/eslint-plugin-jasmine` for the expect error.
* There should be no `covergage` files in your PR.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ c4h_keys.txt
 .idea/vcs.xml
 Project_Default.xml
 .env
+coverage
```
Comment (gngchrs) [inline .gitignore]: minor: extra line

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    ],
+
+    // list of files / patterns to load in the browser
+    files: [
+      './public/lib/angular/angular.js',
+      './public/js/app.js',
+      './public/js/services/*.js',
+      './public/js/controllers/*.js',
+      './public/js/directives.js',
+      './test/mock_test/**/*.js'
```
Comment (gngchrs) [inline karma.conf.js]: minor: path for frontend tests should be `tests/client/**`

File: karma.conf.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    },
+
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    basePath: '',
+
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['jasmine', 'es6-shim'],
```
Comment (gngchrs) [inline karma.conf.js]: minor: we are not using es6-shim this anymore

File: karma.conf.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    plugins: [
+      'karma-chrome-launcher',
+      'karma-jasmine',
+      'karma-coveralls',
+      'karma-coverage',
+      'karma-es6-shim',
```
Comment (gngchrs) [inline karma.conf.js]: ditto: es6-shim

File: package.json [hunk 5]
```diff
@@ -42,6 +42,7 @@
     "gulp-nodemon": "^2.2.1",
     "gulp-sass": "^3.1.0",
     "jade": "~1.11.0",
+    "karma-es6-shim": "^1.0.0",
```
Comment (gngchrs) [inline package.json:45]: ditto: es6-shim

File: package.json [hunk 6]
```diff
@@ -62,20 +63,29 @@
     "view-helpers": "~0.1.3"
   },
   "devDependencies": {
+    "angular": "^1.6.4",
+    "angular-mocks": "^1.6.4",
+    "angular-ui-router": "^0.4.2",
```
Comment (gngchrs) [inline package.json]: minor: I'm not sure why angular stuff is here, coz I see that in your karma config, you reference `public/lib/angular`, but angular being here means it's installed in the `node_modules` folder. These dependencies should probably be in `bower.json`.

**Comment by gngchrs** [comment]
fix merge conflicts and reopen

**Review by Ewanjiru — APPROVED** [review]
merge conflicts all resolved.

**Review by CeciliaWanjiku — APPROVED** [review]

**Pull request closed without merging**